javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/AbstractTest.java

```diff
@@ -45,6 +45,15 @@ protected boolean isJava9() {
         return System.getProperty("java.version").startsWith("9.");
     }
 
+    protected boolean isJava10() {
+        return System.getProperty("java.version").startsWith("10.");
+    }
+
+    protected boolean isJavaVersionAbove9() {
+        String jdkVersion = System.getProperty("java.version");
+        return Integer.parseInt(jdkVersion.substring(0,jdkVersion.indexOf('.'))) >= 9;
```

Comment:
Nitpick: add a space after the comma.

Also, we should consider that the format of the version could change in the future so we could get something that is not a parsable integer.

Reply:
That's a good thought, but as per oracle, for next few years they are going with numeric's like 11,12,13 and so on.. I see there is another problem with these tests. As long as they keep adding methods to String.class these tests are going to break. Rather than asserting on JDK built classes, its better if we rely on java parser classes. Made changes for both of these comments, please check PR (https://github.com/javaparser/javaparser/pull/1672)